### PR TITLE
set useragent

### LINKED ISSUE #647: image promoter should set a useragent
#### What would you like to be added:

Ideally, the image promoter would have a unique useragent so we can identify what requests it's sending to the registry (in particular, to debug why you're getting 429s).

#### Why is this needed:

See https://github.com/kubernetes-sigs/promo-tools/issues/637#issuecomment-1286180731

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

- set useragent

#### Which issue(s) this PR fixes:

Fixes #647 

/assign @saschagrunert @puerco @jeremyrickard @Verolop 
cc @BenTheElder @jonjohnsonjr 

#### Does this PR introduce a user-facing change?

```release-note
set useragent
```

### CODE REVIEW

File: internal/legacy/dockerregistry/inventory.go [hunk 1]
```diff
@@ -839,11 +839,11 @@ func (sc *SyncContext) recordFoundTags(
...
-//  - gcr.io is the registry
-//  - gcr.io/google-containers is the toplevel repository (or "root" repo)
-//  - gcr.io/google-containers/foo is a child repository
-//  - gcr.io/google-containers/foo/bar is a child repository
-//  - gcr.io/google-containers/foo/bar/baz is a child repository
+//   - gcr.io is the registry
+//   - gcr.io/google-containers is the toplevel repository (or "root" repo)
+//   - gcr.io/google-containers/foo is a child repository
+//   - gcr.io/google-containers/foo/bar is a child repository
+//   - gcr.io/google-containers/foo/bar/baz is a child repository
```
Comment (cpanato) [inline internal/legacy/dockerregistry/inventory.go:846]: this came from go fmt 🤷

File: internal/legacy/dockerregistry/inventory.go [hunk 2]
```diff
@@ -1752,6 +1752,7 @@ func (sc *SyncContext) Promote(
 
 					opts := []crane.Option{
 						crane.WithAuthFromKeychain(gcrane.Keychain),
+						crane.WithUserAgent("k8s-image-promoter"),
```
Comment (puerco) [inline internal/legacy/dockerregistry/inventory.go:1755]: Can we const this away in types/image?

```suggestion
						crane.WithUserAgent(image.UserAgent),
```

**Review by puerco — COMMENTED** [review]
Thanks Carlos, one small nit:

File: internal/legacy/dockerregistry/inventory.go (hunk 2 above)
Comment (puerco) [reply internal/legacy/dockerregistry/inventory.go:1755]: And can we name it `kpromo`  :pray:

Comment (cpanato) [reply internal/legacy/dockerregistry/inventory.go]: done, thanks for the feedback

**Review by BenTheElder — COMMENTED** [review]
should we add `-A` to the curl call?

**Comment by jonjohnsonjr** [comment]
> should we add `-A` to the curl call?

Similarly, it would be good to plumb something for the useragent down into the signature code.

**Review by saschagrunert — APPROVED** [review]

**Comment by cpanato** [comment]
> should we add `-A` to the curl call?

which curl call? 🤔

**Comment by BenTheElder** [comment]
https://github.com/kubernetes-sigs/promo-tools/blob/5e139353c19b2fe0b0a3c53e4f31a74262a0a2b4/internal/legacy/dockerregistry/registry/registry.go#L235

on my TODO to file a bug, better answer is to switch to net/http here too but ... too many things going on

**Pull request merged**